### PR TITLE
add lua script support

### PR DESCRIPTION
**Pull request by zreigz**
This PR add support for the Lua script. The script is stored in the `CustomHealth` object.
```
apiVersion: deployments.plural.sh/v1alpha1
kind: CustomHealth
metadata:
  name: test
  namespace: default
spec:
  script: |
    healthStatus = {}
    if Obj.status ~= nil then
      local ready = "Ready"
      if statusConditionExists(Obj.status, ready) then
        healthStatus = {
          status="Progressing"
        }
        if isStatusConditionTrue(Obj.status, ready) then
          healthStatus = {
            status="Healthy"
          }
        end
      end
    end

```

For the input use `Obj` for the output:
```
healthStatus = {
            status="Healthy"
            message="hello"
}
```

### CODE REVIEW

File: pkg/lua/funcs.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type Status struct {
+	Conditions []metav1.Condition
+}
+
+func isStatusConditionExist(s map[string]interface{}, condition string) bool {
```
Comment (michaeljguarino) [inline pkg/lua/funcs.go:13]: perhaps rename to `statusConditionExists`

File: pkg/lua/lua.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	}
+
+	return output, nil
+
+}
+
+func GetFuncMap() template.FuncMap {
+	funcs := sprig.TxtFuncMap()
+	funcs["isStatusConditionTrue"] = isStatusConditionTrue
+	funcs["isStatusConditionExist"] = isStatusConditionExist
```
Comment (michaeljguarino) [inline pkg/lua/lua.go:47]: rename will need to propagate here

File: pkg/lua/funcs.go (hunk 1 above)
Comment (zreigz) [reply pkg/lua/funcs.go]: done

**Comment by michaeljguarino** [comment]
Could we also rename the crd to something like `CustomHealthStatus`?  Feel like that might be a bit more clear for users

File: cmd/main.go [hunk 3]
```diff
@@ -98,6 +105,21 @@ func main() {
 		os.Exit(1)
 	}
 
+	if err = (&controller.HealthConvertReconciler{
+		Client: mgr.GetClient(),
+		Scheme: mgr.GetScheme(),
+		Agent:  a,
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "HealthConvert")
+		os.Exit(1)
```
Comment (michaeljguarino) [inline cmd/main.go:114]: this is a bit different practice, but for now I think we shouldn't exit 1 here as there could be annoying races with the crds being created that we don't anticipate and it's not good for the agent to go down entirely.

If we could drop an error log instead that'd be safer for sure.

**Comment by zreigz** [comment]
> CustomHealthStatus

The CRD also has a Status so it will look strange `CustomHealthStatusStatus` so I will change to `CustomHealth`

File: cmd/main.go (hunk 3 above)
Comment (zreigz) [reply cmd/main.go]: done

**Comment by zreigz** [comment]
@michaeljguarino PTAL

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**